app/src/main/java/nz/org/cacophony/birdmonitor/TestRecordFragment.java

```diff
@@ -177,4 +192,124 @@ public void onReceive(Context context, Intent intent) {
         }
     };
 
+    private boolean requestPermissions(Context context){
+        // If Android OS >= 6 then need to ask user for permission to Write External Storage, Recording, Location
+//        https://developer.android.com/training/permissions/requesting.html
+
+        boolean allPermissionsAlreadyGranted = true;
+
+        if (ContextCompat.checkSelfPermission(context,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+
+            allPermissionsAlreadyGranted = false;
+
+            //https://stackoverflow.com/questions/35989288/onrequestpermissionsresult-not-being-called-in-fragment-if-defined-in-both-fragm
+
+            requestPermissions(new String[]{
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE}, PERMISSION_WRITE_EXTERNAL_STORAGE);
+
+        }
+
+        if (ContextCompat.checkSelfPermission(context,
+                Manifest.permission.RECORD_AUDIO)
+                != PackageManager.PERMISSION_GRANTED) {
+
+            allPermissionsAlreadyGranted = false;
+
+            requestPermissions(new String[]{
+                    Manifest.permission.RECORD_AUDIO}, PERMISSION_RECORD_AUDIO);
+
+        }
+
+        if (ContextCompat.checkSelfPermission(context,
+                Manifest.permission.ACCESS_FINE_LOCATION)
+                != PackageManager.PERMISSION_GRANTED) {
+
+            allPermissionsAlreadyGranted = false;
+
+            requestPermissions(new String[]{
+                    Manifest.permission.ACCESS_FINE_LOCATION}, PERMISSION_LOCATION);
+
+        }
+
+        return allPermissionsAlreadyGranted;
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions,
+                                           int[] grantResults) {
+        // BEGIN_INCLUDE(onRequestPermissionsResult)
+        if (requestCode == PERMISSION_WRITE_EXTERNAL_STORAGE) {
+            // Request for camera permission.
```

Comment:
Incorrect comment. Given this is pretty clear, there's probably no comment required.